frontend/lib/norent/components/subscribe.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+import { Trans, t } from "@lingui/macro";
+import { li18n } from "../../i18n-lingui";
+import { StaticImage } from "../../ui/static-image";
+import { getImageSrc } from "../homepage";
+
+type SubscribeProps = {};
+
+type SubscribeState = {
+  email: string;
+  success: boolean;
+  response: string;
+};
+
+class Subscribe extends React.Component<SubscribeProps, SubscribeState> {
+  constructor(props: SubscribeProps) {
+    super(props);
+    this.state = {
+      email: "",
+      success: false,
+      response: "",
+    };
+  }
+
+  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ email: e.target.value });
+  };
+
+  handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const email = this.state.email || null;
+    const locale = li18n.language;
+
+    if (!email) {
+      this.setState({
+        response: li18n._(t`Please enter an email address!`),
+      });
+      return;
+    }
+
+    fetch("/mailchimp/subscribe", {
+      method: "POST",
+      body: `email=${encodeURIComponent(
+        email
+      )}&language=${locale}&source=norent`,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    })
+      .then((result) => result.json())
+      .then((result) => {
+        if (result.status === 200) {
+          this.setState({
+            success: true,
+            response: li18n._(t`All set! Thanks for subscribing!`),
+          });
+        } else if (result.errorCode === "INVALID_EMAIL") {
+          this.setState({
+            response: li18n._(t`Oops! That email is invalid.`),
+          });
+        } else {
+          window &&
+            window.Rollbar &&
+            window.Rollbar.error(
+              `Mailchimp email signup responded with error code ${result.errorCode}.`
+            );
+          this.setState({
+            response: li18n._(
+              t`Oops! A network error occurred. Try again later.`
```

Comment:
We've actually got some text _extremely_ similar to this in `app.tsx` that says `Unfortunately, a network error occurred. Please try again later.`.  Maybe we should reuse that to reduce the amount of translation work that needs to be done (and so our content is more consistent)?  You could either reuse the text verbatim here, or I guess you could factor out a separate module that centralizes it one place.

Reply:
Great! I'm actually gonna make the TP consistent with our other products and replaces these instances with the "Oops!..." version.